src/braket/experimental/autoqasm/program/program.py

```diff
@@ -313,6 +348,16 @@ def get_free_parameters(self) -> list[oqpy.FloatVar]:
         """Return a list of named oqpy.Vars that are used as free parameters in the program."""
         return list(self._free_parameters.values())
 
+    def add_io_declarations(self) -> None:
+        """Add input and output declaration statements to the program."""
+        free_parameters = self.get_free_parameters()
+        root_oqpy_program = self.get_oqpy_program(scope=ProgramScope.MAIN)
+        for parameter in free_parameters[::-1]:
```

Comment:
[nit] to avoid an extra variable declaration and a list copy:
```suggestion
        root_oqpy_program = self.get_oqpy_program(scope=ProgramScope.MAIN)
        for parameter in reversed(self.get_free_parameters()):
```